third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1777,6 +1777,12 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
                 if let StructLayout::Singleton(fields, _is_positional) = &entry.layout {
                     et.enter_scope();
                     for f in fields.values() {
+                        if et.env().language_version.is_at_least(LanguageVersion::V2_0)
```

Comment:
Can you pull this language version lookup out of the loop?  I don't think Rust will be able to CSE it.  It also could be clearer to give it a name.

Reply:
Also, can you add a brief comment to explain what is happening here?  IIUC,
```
// In Aptos Move 2.0 and above, field `self` is omitted from local bindings so `self` can be used to refer to `self` parameter.
```

Reply:
done